File: tasks/implementations/go/tasks/sort_integers.go
package main

import (
    "bufio"
    "fmt"
    "os"
    "path/filepath"
    "sort"
    "strconv"
)

func RunSortIntegers(size string, fixturesRoot string) {
    path := filepath.Join(fixturesRoot, "generated", "sort", size+".txt")
    file, _ := os.Open(path)
    defer file.Close()
    scanner := bufio.NewScanner(file)
    values := make([]int, 0)
    for scanner.Scan() {
        n, _ := strconv.Atoi(scanner.Text())
        values = append(values, n)
    }
    sort.Ints(values)
    total := 0
    for i := 0; i < 100 && i < len(values); i++ { total += values[i] }
    fmt.Println(total)
}

File: tasks/implementations/cpp/tasks/sort_integers.cpp
#include <algorithm>
#include <fstream>
#include <iostream>
#include <string>
#include <vector>

void run_sort_integers(const std::string& size, const std::string& fixturesRoot) {
    std::ifstream file(fixturesRoot + "/generated/sort/" + size + ".txt");
    std::vector<int> values;
    int value;
    while (file >> value) values.push_back(value);
    std::sort(values.begin(), values.end());
    long long total = 0;
    for (int i = 0; i < 100 && i < static_cast<int>(values.size()); ++i) total += values[i];
    std::cout << total << std::endl;
}
